NativeApp/src/components/Booking/container.js

```diff
@@ -0,0 +1,101 @@
+import React, { Component } from 'react';
+import { Alert } from 'react-native';
+import { PropTypes as PT } from 'prop-types';
+import moment from 'moment';
+import { userProfile } from '../../utilities/currentUser';
+import { requestHolidays } from '../../services/holidayService';
+
+
+export default Container => class extends Component {
+  static propTypes = {
+    navigation: PT.shape({
+      navigate: PT.fun,
+    }).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      booking: {
+        startDate: '',
+        endDate: '',
+        halfDay: false,
+      },
+      user: {},
+    };
+  }
+
+  componentDidMount() {
+    const { navigation } = this.props;
+    const chosenDate = navigation.getParam('date', '');
+
+    userProfile()
+      .then(user => this.setState({ user }));
+
+    this.setState({
+      booking: {
+        startDate: chosenDate,
+        endDate: chosenDate,
+      },
+    });
+  }
+
+  changeStartDate = (date) => {
+    const formatDate = moment(date).format('YYYY-MM-DD');
+    this.setState(prevState => ({
```

Comment:
Is there any particular benefit to using `prevState` rather than `this.state`? Just a query as I've never seen `setState` used this way before!

Reply:
This had to do with changing the date picker. I have two, start and end date pickers. Therefore each receive their own state for chosen date. I have to update the state when the date picker is changed. Problem was either the start date or end date state was disappearing on change.

The `prevstate` basically allows me to bring in prev state and I can update either end date or start date without erasing them.